src/renderer/components/settings/components.tsx

```diff
@@ -151,9 +155,11 @@ const CacheSetting = memo(({ setting, value, setValue }: SettingsProps<'cache'>)
                             locationPromise
                                 .then(async (cacheLocation) => {
                                     try {
-                                        await access(cacheLocation);
+                                        await ipcRenderer.invoke('fs-access', cacheLocation);
```

Comment:
This relies on `access` throwing an error. I don't think errors are ICP compatible, so how does this work now?

Reply:
Based on s a quick Google search it seems it handles errors fine now, but I'll try this out later to confirm

Reply:
can confirm it does properly pass through errors.